var/spack/repos/builtin/packages/vasp/package.py

```diff
@@ -29,6 +30,8 @@ class Vasp(MakefilePackage):
              tag='V1.0',
              when='+vaspsol')
 
+    variant('openmp', default=False,
```

Comment:
The `openmp` variant has effect only with `aocc` if I read this correctly. Can you extend the handling of this option to other compilers too?

Reply:
@amd-toolchain-support VASP is proprietary software and currently I don't have access to it to perform a build and test drive the recipe. I'll be iterating on this PR to check idioms etc. but it would be better if somebody that has a license could verify a few builds.

@nrichart Are you still deploying VASP for users at EPFL? If so, would you have time / be willing to check this recipe builds fine?

Reply:
Hi @alalazo, `openmp`  variant is valid for other compilers too, but we have verified it only for AOCC.

Please find the `openmp` options for other compilers 
https://www.vasp.at/wiki/index.php/Hybrid_MPI/OpenMP_parallelization

Reply:
Sure. What I mean is that with the current recipe if somebody asks for `vasp+openmp %gcc` what effectively gets built is a version _without_ OpenMP which is unexpected. Can you fix that and enable OpenMP builds for other compilers too?

Reply:
got it, as per the `makefiles` available as below for GNU and other compilers, I have made few changes to the recipe, please checkout latest push

```
makefile.include.linux_gnu
makefile.include.linux_gnu_omp
makefile.include.linux_intel
makefile.include.linux_intel_omp
makefile.include.linux_intel_serial
makefile.include.linux_nv
makefile.include.linux_nv_acc
makefile.include.linux_nv_omp
makefile.include.linux_pgi
makefile.include.linux_pgi_acc
makefile.include.linux_pgi_omp
```

Reply:
@alalazo, yes we are still deploying VASP at EPFL. I can try to build this recipe.

Reply:
After saying yes I realized, I only have access to the version 5.4.4 at the moment. It compiles and runs fine with `intel ^intel-mkl`.

I have an issue with `gcc` and `intel ^netlib-scalapack` but it is not linked to this PR. (we use an old local recipe so we never realized)
The problem seams to come from the part on `scalapack` not handled the same if the compiler is not `%aocc` or `%nvhpc`

Would it make sense to change `filter_file('^SCALAPACK*$', '', 'makefile.include')` line 138 which in any case is not a proper regexp `.` missing before the `*` by the filter that is in the case of `%aocc`
It worked for me on the version 5.4.4